types/google-protobuf/index.d.ts

```diff
@@ -109,8 +109,8 @@ export abstract class Message {
   static equals(m1: Message, m2: Message): boolean;
   static compareExtensions(extension1: {}, extension2: {}): boolean;
   static compareFields(field1: any, field2: any): boolean;
-  cloneMessage(): Message;
-  clone(): Message;
+  cloneMessage<T extends Message>(): T;
+  clone<T extends Message>(): T;
```

Comment:
I don't think this change is needed at all since TypeParameters do not affect anything. Person who is expecting to run `clone<SomeMessage>()` can always instead write it as `clone() as SomeMessage`

Reply:
@sheetalkamat?